src/api.ts

```diff
@@ -0,0 +1,559 @@
+      prepared
+    )}`;
+    return [template, serviceCall];
+  } else {
+    const obj = stmt.expr;
+    if (!isObjectLiteralExpr(obj)) {
+      throw new Error(
+        "Expected response function body to return an object literal"
+      );
+    }
+    const template = `#set($inputRoot = $input.path('$'))\n${stringify(
+      inner(obj)
+    )}`;
+    return [template, undefined];
+  }
+
+  function inner(node: FunctionlessNode): any {
+    if (
+      isBooleanLiteralExpr(node) ||
+      isNumberLiteralExpr(node) ||
+      isStringLiteralExpr(node) ||
+      isNullLiteralExpr(node)
+    ) {
+      return node.value;
+    } else if (isArrayLiteralExpr(node)) {
+      return node.children.map(inner);
+    } else if (isObjectLiteralExpr(node)) {
+      return Object.fromEntries(
+        node.properties.map((prop) => {
+          switch (prop.kind) {
+            case "PropAssignExpr":
+              return [inner(prop.name), inner(prop.expr)];
+            case "SpreadAssignExpr":
+              throw new Error("TODO: support SpreadAssignExpr");
+          }
+        })
+      );
+    } else if (isArgument(node)) {
+      return inner(node);
+    } else if (isPropAccessExpr(node)) {
+      // ignore the function param name, we'll replace it with the VTL
+      // mapping template inputs
+      const path = pathFromFunctionParameter(node)?.slice(1);
+
+      if (path) {
+        if (location === "response") {
+          const ref: Ref = {
+            __refType: node.type! as any,
+            value: `$inputRoot.${path.join(".")}`,
+          };
+          return ref;
+        } else {
+          const [paramLocation, ...rest] = path;
+
+          let prefix;
+          switch (paramLocation) {
+            case "body":
+              prefix = "$inputRoot";
+              break;
+            case "pathParameters":
+              prefix = "$input.params().path";
+              break;
+            case "queryStringParameters":
+              prefix = "$input.params().querystring";
+              break;
+            case "headers":
+              prefix = "$input.params().header";
+              break;
+            default:
+              throw new Error("Unknown parameter type.");
+          }
+
+          const param = `${prefix}.${rest.join(".")}`;
+
+          const ref: Ref = { __refType: node.type! as any, value: param };
+          return ref;
+        }
+      }
+      return `${inner(node.expr)}.${node.name};`;
+    } else if (isTemplateExpr(node)) {
+      // TODO: not right, compare to vtl.ts
+      return inner(node.exprs[0]);
+    }
+
+    throw new Error(`Unsupported node type: ${node.kind}`);
+  }
+}
+
+// These represent variable references and carry the type information.
+// stringify will serialize them to the appropriate VTL
+// e.g. if `request.pathParameters.id` is a number, we want to serialize
+// it as `$input.params().path.id`, not `"$input.params().path.id"` which
+// is what JSON.stringify would do
+type Ref =
```

Comment:
I felt dirty writing this. hopefully someone has a better way

Might just go all the way and write a simple JSON ADT with an extra Ref node type